### DISCUSSION TITLE
Add "short words" for word based movements

### DISCUSSION THREAD

**Discussion by ModProg** [helix-editor/helix#11619] [Ideas] [answered]
Currently helix has regular words e.g. `move_next_word_end`/`e`  and long words `move_next_long_word_end`/`E´. I'd like to propose "short words", i.e. parts of words that are e.g. camelCase or snake_case.

The behavior I'd expect for the short word movement compared to regular and long (`|` is the current cursor position):

`th|isIsAWord-otherWord separateLongWord`

1. `move_next_short_word_end`: `this|IsAWord-otherWord separateLongWord`
2. `move_next_word_end`: `thisIsAWord|-otherWord separateLongWord`
3. `move_next_long_word_end`: `thisIsAWord-otherWord| separateLongWord`

This would be useful when e.g. changing parts of a test function name you pasted.

I'm unsure if there is a good default keymap to give this if any, would have proposed something like `A-e` but that is already occupied. This could also be a binding that is up to the user to assign.

**Comment by kirawi** [comment, marked as answer]
https://github.com/helix-editor/helix/pull/8147

**Reply by ModProg** [reply]
Thanks, for some reason I was really convinced on `short` being the right word :D
